### PR TITLE
Implement Subtitles

### PR DESCRIPTION
**Pull request by arj119**
### Clubhouse Ticket
[ch320](https://app.clubhouse.io/manim-dsl/story/320/implement-subtitles-on-compiler-backend)

### Description

To implement subtitles fully (building on #99) 

Fixes # (issue) 

### Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe tests added as well.
 
Semantic tests, ast generation tests

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/main/kotlin/com/manimdsl/linearrepresentation/Objects.kt [hunk 1]
```diff
@@ -13,17 +13,18 @@ sealed class MObject : ManimInstr() {
     abstract val shape: Shape
 }
 
-sealed class ShapeWithBoundary(open val uid: String) : MObject() {
```
Comment (mayanksharma3) [inline src/main/kotlin/com/manimdsl/linearrepresentation/Objects.kt:16]: Keep the sealed class and get rid of the ShapeWithBoundary interface, I think it makes more sense for it to extend MObject() as they all have Shapes underneath

File: src/main/kotlin/com/manimdsl/runtime/VirtualMachine.kt [hunk 2]
```diff
@@ -284,9 +280,51 @@ class VirtualMachine(
...
+                val position = stylesheet.getPosition(dsUID)
+                val boundaries = if (position == null) emptyList() else getBoundaries(position)
+                subtitleBlockVariable = SubtitleBlock(
+                    variableNameGenerator,
+                    runtime = animationSpeeds.first(),
+                    uid = dsUID,
+                    boundary = boundaries,
+                    textColor = stylesheet.getSubtitleStyle().textColor,
+                    duration = duration
+                ) // TODO(aesthetics stuff)
```
Comment (claralebbos) [inline src/main/kotlin/com/manimdsl/runtime/VirtualMachine.kt:316]: what needs to be done here?

**Comment by mayanksharma3** [comment]
Is the subtitle positionable from the Stylesheet too?

File: src/main/kotlin/com/manimdsl/runtime/VirtualMachine.kt (hunk 2 above)
Comment (arj119) [reply src/main/kotlin/com/manimdsl/runtime/VirtualMachine.kt:316]: OH I did this lol I will delete

**Review by mayanksharma3 — APPROVED** [review]

**Pull request merged**